.drone/pipelines/default/steps/test/cats_internetless.sh

```diff
@@ -0,0 +1,148 @@
+
+    if [[ $enable == 1 ]]; then
+	# Complaint wrong. The echo generates a traling newline the `blue` doesn't.
+	# shellcheck disable=SC2005
+	echo "$(blue "Configure cluster network: Deny egress external")"
+	# enable isolation
+	# ingress - allows all incoming traffic
+	# egress  - allows dns traffic anywhere
+	#         - allows traffic to all ports, pods, namespaces
+	#           (but no whitelisting of external ips!)
+	# references
+	# - BASE = https://github.com/ahmetb/kubernetes-network-policy-recipes
+	# - (BASE)/blob/master/02a-allow-all-traffic-to-an-application.md
+	# - (BASE)/blob/master/14-deny-external-egress-traffic.md
+	# - See also https://www.youtube.com/watch?v=3gGpMmYeEO8 (31min)
+	#   - Egress info wrt disallow external see 17:20-17:52
+	#
+	# __ATTENTION__
+	# Requires a networking plugin to enforce, else ignored
+	# (if not directly supported by platform)
+	# - Example plugins: Calico, WeaveNet, Romana
+	#
+	# GKE: Uses Calico, Use `--enable-network-policy` when
+	# creating a cluster (`gcloud`).
+	# Minikube needs special setup.
+	# KinD used by our Drone setup may have support.
+
+	cat <<EOF | kubectl apply -f -
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: cats-internetless
+  namespace: ${KUBECF_NAMESPACE}
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+  ingress:
+  - {}
+  egress:
+    - ports:
+      - port: 53
+        protocol: UDP
+      - port: 53
+        protocol: TCP
+    - to:
+      - namespaceSelector: {}
+EOF
+	# For debugging, show what kube thinks of it.
+	kubectl describe networkpolicies \
+		--namespace "${KUBECF_NAMESPACE}" \
+		cats-internetless
+    else
+	# shellcheck disable=SC2005
+	echo "$(blue "Configure cluster network: Full access")"
+	# disable isolation
+	kubectl delete networkpolicies \
+		--namespace "${KUBECF_NAMESPACE}" \
+		cats-internetless
+    fi
+}
+
+configure_cats_suites() {
+    # shellcheck disable=SC2005
+    echo "$(blue "Configure CATS: =internetless")"
+
+    # External script. Variant of the script `deploy/kubecf.sh`
+    # modified to limit the suites used by the cats to `internetless`
+    # alone.
+    #
+    # External __by choice__. Make it easier in the future to
+    # consolidate with the original script and use parameters,
+    # templating, whatever to choose mode on call.
+    #
+    # || true catches that reconfiguration generates errors of the
+    # form ``invalid: spec.clusterIP: Invalid value: "": field is immutable``.
+    # These errors do not seem to prevent us from running the CATS again.
+    # And the redirection hides the output from that.
+
+    { ".drone/pipelines/default/runtime/cats_kubecf.sh" >& /dev/null; } || true
+    wait_for_ig_job || exit 1
+}
+
+deconfigure_cats_suites() {
+    # shellcheck disable=SC2005
+    echo "$(blue "Configure CATS: defaults")"
+
+    # Undo the limited configuration of the cats.
+    { ".drone/pipelines/default/runtime/decats_kubecf.sh" >& /dev/null; } || true
```

Comment:
I think `decats_kubecf` is a misnomer. I only understood it after I read the whole logic of this file. `configure_cats_suites` is also a misnomer. Even though it's "configuring" CATS suites, it's very specific to the `internetless` suite.

I don't think it's the responsibility of the internetless suite to "deconfigure" CATS. Just leave the way it exited. If another configuration is needed in the future, it should be the responsibility of the next CI job or pipeline.

Reply:
So the actions you are asking for are:
- Rename `configure_cats_suites` to `configure_cats_for_internetless`
- Drop the entire `decats_...` function and script.
right ?
If yes, accepted.

Reply:
Correct

Reply:
Hopefully resolved with commit af340e6f707d501c3928efc88ed42389180f7845